src/Gui/DlgPreferencesImp.cpp

```diff
@@ -312,6 +312,21 @@ void DlgPreferencesImp::activateGroupPage(const QString& group, int index)
     }
 }
 
+/**
+ * Returns the group name \a group and position \a index of the active page.
+ */
+void DlgPreferencesImp::activeGroupPage(QString& group, int& index) const
+{
+    int row = ui->listBox->currentRow();
+    auto item = ui->listBox->item(row);
+    auto tabWidget = static_cast<QTabWidget*>(ui->tabWidgetStack->widget(row));

```

Comment:
Clang-tidy complains about the static_cast for me, it wants a dynamic_cast (which is in line with your check for null in the code below -- static_cast won't usually fail with a nullptr, if it isn't the right cast then we're in Undefined Behavior territory).